gadget/partial.go

```diff
@@ -0,0 +1,129 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2023 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package gadget
+
+import "fmt"
+
+// ApplyInstallerVolumesToGadget takes the volume information returned
+// by the installer and applies it to the laid out volumes for
+// properties partially defined. After that it checks that the gadget
+// is now fully specified.
+func ApplyInstallerVolumesToGadget(installerVols map[string]*Volume, lovs map[string]*LaidOutVolume) error {
+	for volName, lov := range lovs {
+		if len(lov.Partial) == 0 {
+			continue
+		}
+
+		insVol := installerVols[volName]
+		if insVol == nil {
+			return fmt.Errorf("installer did not provide information for volume %q", volName)
+		}
+
+		// TODO: partial structure, as it is not clear what will be possible when set
+
+		if lov.HasPartial(PartialSchema) {
+			if insVol.Schema == "" {
+				return fmt.Errorf("installer did not provide schema for volume %q", volName)
+			}
+			lov.Schema = insVol.Schema
```

Comment:
lov.Schema could be already be set?

Reply:
should we make it an error then to have partial:[schema] but schema set in the volume?

Reply:
I have added the check in gadget.go